components/brave_wallet_ui/common/async/wallet_async_handler.ts

```diff
@@ -92,6 +92,35 @@ async function refreshWalletInfo (store: Store) {
   const hardwareAccounts = await apiProxy.keyringController.getHardwareAccounts()
   result.accountInfos = [...result.accountInfos, ...hardwareAccounts.accounts]
 
+  // Get/Set selectedAccount
+  if (result.isWalletCreated) {
+
+    // Get selectedAccountAddress
+    const getSelectedAccount = await apiProxy.keyringController.getSelectedAccount()
+    const selectedAddress = getSelectedAccount.address
+
+    // Fallback account address if selectedAccount returns null
+    const fallbackAddress = result.accountInfos[0].address
+
+    // If selectedAccount is null will setSelectedAccount to fallback address
+    if (!selectedAddress) {
+      await apiProxy.keyringController.setSelectedAccount(fallbackAddress)
+      result.selectedAccount = fallbackAddress
+    } else {
+      // If a user has already created an wallet but then chooses to restore
+      // a different wallet, getSelectedAccount still returns the previous wallets
+      // selected account.
+      // This check looks to see if the returned selectedAccount exist in the accountInfos
+      // payload, if not it will setSelectedAccount to the fallback address
+      if (!result.accountInfos.find((account) => account.address.toLowerCase() === selectedAddress?.toLowerCase())) {
+        result.selectedAccount = fallbackAddress
+        await apiProxy.keyringController.setSelectedAccount(fallbackAddress)
```

Comment:
We can eventually refactor some of this fallback address logic in the controller when we have a bit more time. I posted this to track that: https://github.com/brave/brave-browser/issues/18528

Reply:
Perfect!!